SHA3.cs

```diff
@@ -0,0 +1,191 @@
+    public static RSAEncryptionPadding OaepSHA3_512 { get; }
+}
+
+public abstract partial class SHA3_256 : HashAlgorithm {
+    public const int HashSizeInBits = 256;
+    public const int HashSizeInBytes = 32;
+
+    protected SHA3_256();
+
+    public static bool IsSupported { get; }
+
+    public static new SHA3_256 Create();
+
+    public static byte[] HashData(byte[] source);
+    public static byte[] HashData(ReadOnlySpan<byte> source);
+    public static int HashData(ReadOnlySpan<byte> source, Span<byte> destination);
+    public static bool TryHashData(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesWritten);
+
+    public static byte[] HashData(Stream source);
+    public static int HashData(Stream source, Span<byte> destination);
+    public static ValueTask<int> HashDataAsync(Stream source, Memory<byte> destination, CancellationToken cancellationToken = default);
+    public static ValueTask<byte[]> HashDataAsync(Stream source, CancellationToken cancellationToken = default);
+}
+
+
+public abstract partial class SHA3_384 : HashAlgorithm {
+    public const int HashSizeInBits = 384;
+    public const int HashSizeInBytes = 48;
+
+    protected SHA3_384();
+
+    public static bool IsSupported { get; }
+
+    public static new SHA3_384 Create();
+
+    public static byte[] HashData(byte[] source);
+    public static byte[] HashData(ReadOnlySpan<byte> source);
+    public static int HashData(ReadOnlySpan<byte> source, Span<byte> destination);
+    public static bool TryHashData(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesWritten);
+
+    public static byte[] HashData(Stream source);
+    public static int HashData(Stream source, Span<byte> destination);
+    public static ValueTask<int> HashDataAsync(Stream source, Memory<byte> destination, CancellationToken cancellationToken = default);
+    public static ValueTask<byte[]> HashDataAsync(Stream source, CancellationToken cancellationToken = default);
+}
+
+public abstract partial class SHA3_512 : HashAlgorithm {
+    public const int HashSizeInBits = 512;
+    public const int HashSizeInBytes = 64;
+
+    protected SHA3_512();
+
+    public static bool IsSupported { get; }
+
+    public static new SHA3_512 Create();
+
+    public static byte[] HashData(byte[] source);
+    public static byte[] HashData(ReadOnlySpan<byte> source);
+    public static int HashData(ReadOnlySpan<byte> source, Span<byte> destination);
+    public static bool TryHashData(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesWritten);
+
+    public static byte[] HashData(Stream source);
+    public static int HashData(Stream source, Span<byte> destination);
+    public static ValueTask<int> HashDataAsync(Stream source, Memory<byte> destination, CancellationToken cancellationToken = default);
+    public static ValueTask<byte[]> HashDataAsync(Stream source, CancellationToken cancellationToken = default);
+}
+
+
+public partial class HMACSHA3_256 : HMAC {
```

Comment:
Like SHA-3, the HMACSHA-3 classes follow our existing HMACSHA-2 classes, with the same exceptions.